libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -460,3 +462,55 @@ void ContractLevelChecker::checkLibraryRequirements(ContractDefinition const& _c
 		if (!var->isConstant())
 			m_errorReporter.typeError(var->location(), "Library cannot have non-constant state variables");
 }
+
+void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _contract)
+{
+	if (_contract.sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2))
+		return;
+
+	SecondarySourceLocation errors;
+
+	// interfaceFunctionList contains all inherited functions as well
+	for (auto const& func: _contract.interfaceFunctionList())
+	{
+		solAssert(func.second->hasDeclaration(), "Function has no declaration?!");
+
+		if (!func.second->declaration().sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2))
+			continue;
+
+		bool nextFunction = false;
+
+		auto const& currentLoc = func.second->declaration().location();
+
+		for (auto const& paramType: func.second->parameterTypes())
+			if (!typeSupportedByOldABIEncoder(*paramType, false))
```

Comment:
Please assert that this is not a library.

Reply:
Can't you inherit from a library?

Reply:
No - but we might change that in the future. The reason I would like the assert is because you are using `false` here, which means "not a library function" and this should be asserted.

Reply:
Should I assert all base contracts or just the current one?

Reply:
I guess the current one is fine.

Reply:
Hmm the assert fails. While there is already an error added for when it is a library, it is not a fatal error, so execution continues. I could add the condition that errors must be added, too?

Reply:
Perhaps check that there are at least two contracts in the linearized base class list?